JOSESwift/Sources/JWE.swift

```diff
@@ -72,10 +72,19 @@ public struct JWE {
     /// - Throws: `JOSESwiftError` if any error occurs while encrypting.
     public init<KeyType>(header: JWEHeader, payload: Payload, encrypter: Encrypter<KeyType>) throws {
         self.header = header
+        if header.zip != nil && header.zip != CompressionAlgorithm.deflate.rawValue {
```

Comment:
In my opinion, here and below it would be handy to be able to switch on `zip` instead of doing string comparisons.